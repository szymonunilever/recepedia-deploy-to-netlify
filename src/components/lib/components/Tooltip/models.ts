import { RCTooltip } from 'rc-tooltip';
export interface TooltipProps extends RCTooltip.Props {
  className?: string;
}
