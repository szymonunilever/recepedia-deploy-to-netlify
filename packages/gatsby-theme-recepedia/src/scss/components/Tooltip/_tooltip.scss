.rc-tooltip {
  font-size: 1.4rem;
  background: rgba(97, 97, 97, 0.95);
  padding: 10px;
  border-radius: 5px;
  color: white;
  position: relative;
  @include respondTo(medium) {
    width: 250px;
    transform: translate(-50%, -50%);
  }
}
