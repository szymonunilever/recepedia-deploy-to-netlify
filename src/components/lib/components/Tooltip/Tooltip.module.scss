.tooltip{
  &:global(.rc-tooltip-hidden){
    height: 0;
    visibility: hidden;
  }
}
