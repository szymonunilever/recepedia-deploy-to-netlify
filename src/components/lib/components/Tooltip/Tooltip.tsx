import React from 'react';
import RCTooltip from 'rc-tooltip';
import { TooltipProps } from './models';
import theme from './Tooltip.module.scss';
import cx from 'classnames';

const Tooltip = ({ ...props }: TooltipProps) => {
  const onAligned = () => {
    props.children.ref.current.classList.toggle('active');
  };
  return (
    <RCTooltip
      onVisibleChange={onAligned}
      overlayClassName={cx(props.className, theme.tooltip)}
      {...props}
    />
  );
};
export default Tooltip;
