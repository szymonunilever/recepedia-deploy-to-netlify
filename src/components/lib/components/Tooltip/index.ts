export { default } from './Tooltip';
export { default as Tooltip } from './Tooltip';
export * from './models';
