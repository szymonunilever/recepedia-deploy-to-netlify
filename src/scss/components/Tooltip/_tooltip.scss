.rc-tooltip {
  font-size: 14px;
  background: rgba(97, 97, 97, 0.95);
  padding: 10px;
  border-radius: 5px;
  color: white;
  position: relative;
  @include respondTo(medium) {
    width: 250px;
    transform: translate(-50%, -50%);
  }
  &:before {
    display: none;
    content: '';
    width: 0;
    height: 0;
    position: absolute;
    top: -16px;
    left: 50%;
    transform: translateX(-50%);
    border-top: 8px solid transparent;
    border-bottom: 8px solid rgba(97, 97, 97, 0.95);
    border-right: 8px solid transparent;
    border-left: 8px solid transparent;
    @include respondTo(medium) {
      display: block;
    }
  }
}
