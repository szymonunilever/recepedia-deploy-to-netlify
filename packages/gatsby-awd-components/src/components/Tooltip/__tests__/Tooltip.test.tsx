import { mount, ReactWrapper } from 'enzyme';
import React from 'react';
import Tooltip, { TooltipProps } from '../index';
import toJson from 'enzyme-to-json';

describe('<Tooltip />', () => {
  let wrapper: ReactWrapper;
  // @ts-ignore
  const tooltipProps: TooltipProps = {
    className: 'test-class',
    // children: [],
    overlay: 'some test info',
  };

  beforeEach(() => {
    wrapper = mount(
      <Tooltip {...tooltipProps}>
        <div>test123</div>
      </Tooltip>
    );
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('matches the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
