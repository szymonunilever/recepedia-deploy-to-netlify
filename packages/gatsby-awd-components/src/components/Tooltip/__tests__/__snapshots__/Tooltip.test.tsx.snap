// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> matches the snapshot 1`] = `
<Tooltip
  className="test-class"
  overlay="some test info"
>
  <Tooltip
    align={Object {}}
    arrowContent={null}
    className="test-class"
    destroyTooltipOnHide={false}
    mouseEnterDelay={0}
    mouseLeaveDelay={0.1}
    onVisibleChange={[Function]}
    overlay="some test info"
    overlayClassName="test-class tooltip"
    placement="right"
    prefixCls="rc-tooltip"
    trigger={
      Array [
        "hover",
      ]
    }
  >
    <Trigger
      action={
        Array [
          "hover",
        ]
      }
      afterPopupVisibleChange={[Function]}
      arrowContent={null}
      blurDelay={0.15}
      builtinPlacements={
        Object {
          "bottom": Object {
            "offset": Array [
              0,
              4,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "tc",
              "bc",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "bottomLeft": Object {
            "offset": Array [
              0,
              4,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "tl",
              "bl",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "bottomRight": Object {
            "offset": Array [
              0,
              4,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "tr",
              "br",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "left": Object {
            "offset": Array [
              -4,
              0,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "cr",
              "cl",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "leftBottom": Object {
            "offset": Array [
              -4,
              0,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "br",
              "bl",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "leftTop": Object {
            "offset": Array [
              -4,
              0,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "tr",
              "tl",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "right": Object {
            "offset": Array [
              4,
              0,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "cl",
              "cr",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "rightBottom": Object {
            "offset": Array [
              4,
              0,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "bl",
              "br",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "rightTop": Object {
            "offset": Array [
              4,
              0,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "tl",
              "tr",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "top": Object {
            "offset": Array [
              0,
              -4,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "bc",
              "tc",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "topLeft": Object {
            "offset": Array [
              0,
              -4,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "bl",
              "tl",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "topRight": Object {
            "offset": Array [
              0,
              -4,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "br",
              "tr",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
        }
      }
      className="test-class"
      defaultPopupVisible={false}
      destroyPopupOnHide={false}
      focusDelay={0}
      getDocument={[Function]}
      getPopupClassNameFromAlign={[Function]}
      hideAction={Array []}
      mask={false}
      maskClosable={true}
      mouseEnterDelay={0}
      mouseLeaveDelay={0.1}
      onPopupAlign={[Function]}
      onPopupVisibleChange={[Function]}
      overlay="some test info"
      popup={[Function]}
      popupAlign={Object {}}
      popupClassName="test-class tooltip"
      popupPlacement="right"
      popupStyle={Object {}}
      prefixCls="rc-tooltip"
      showAction={Array []}
    >
      <div
        className="test-class"
        key="trigger"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        test123
      </div>
    </Trigger>
  </Tooltip>
</Tooltip>
`;
